4_de_analysis/nbconverted/0_download_counts.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Download pan-cancer RNA-seq read counts data from UCSC Xena Browser
+# 
+# The other RNA-seq data we downloaded in `0_data_download` contains pre-processed RPKM values. In order to do differential expression analysis, most methods recommend using count data, or something similar such as [RSEM expected counts](https://support.bioconductor.org/p/90672/#90678) (which is what we'll download here).
+# 
+# GDC does not seem to store RNA-seq read counts (that I'm aware of), so we'll download it from the UCSC Xena Browser instead. This data was generated as part of the Pan-Cancer Atlas project so it should apply to the same set of samples.
+
+# In[1]:
+
+
+import pandas as pd
+# from urllib.request import urlretrieve
+
+import sys; sys.path.append('..')
+import config as cfg
+
+cfg.de_data_dir.mkdir(parents=True, exist_ok=True)
+cfg.raw_de_data_dir.mkdir(parents=True, exist_ok=True)
+
+
+# In[2]:
+
+
+base_url = 'https://toil-xena-hub.s3.us-east-1.amazonaws.com/download/'
+filename = 'tcga_gene_expected_count'
+
+url = base_url + filename + '.gz'
+output_filename = cfg.raw_de_data_dir / (filename + '.tsv.gz')
+
+if not output_filename.is_file():
```

Comment:
More for my own curiosity, is there a reason you didn't download all the data at once and then read in from your local dir? Or the opposite, is there a reason you don't just read in the data from online and store it in the dataframe? Why are you doing a combination?

Reply:
Downloading the data is pretty slow (it's a 500MB dataframe) so I don't want to download it each time the script is run, just if it's not already present where the script expects it to be.

This was mostly useful when I was writing the code to preprocess the matrix, since a new user would probably only run the script once anyway, but I'm leaving the check here in case I want to change the preprocessing steps in the future.